### PR TITLE
Changed ContentRange from std::string to Azure::Core::Http::Range.

### PR DESCRIPTION
**Pull request by katmsft**
Fixes https://github.com/Azure/azure-sdk-for-cpp/issues/1347

# Pull Request Checklist

Please leverage this checklist as a reminder to address commonly occurring feedback when submitting a pull request to make sure your PR can be reviewed quickly:

See the detailed list in the [contributing guide](https://github.com/Azure/azure-sdk-for-cpp/blob/master/CONTRIBUTING.md#pull-requests).

- [x] [C++ Guidelines](https://azure.github.io/azure-sdk/cpp_introduction.html)
- [x] Doxygen docs
- [x] Unit tests
- [x] No unwanted commits/changes
- [x] Descriptive title/description
  - [x] PR is single purpose
  - [x] Related issue listed
- [x] Comments in source
- [x] No typos
- [x] Update changelog
- [x] Not work-in-progress
- [x] External references or docs updated
- [x] Self review of PR done
- [x] Any breaking changes?

### CODE REVIEW

File: sdk/storage/azure-storage-files-datalake/src/datalake_file_client.cpp [hunk 1]
```diff
@@ -326,12 +326,8 @@ namespace Azure { namespace Storage { namespace Files { namespace DataLake {
     Models::ReadDataLakeFileResult ret;
     ret.Body = std::move(result->BodyStream);
     ret.HttpHeaders = FromBlobHttpHeaders(std::move(result->HttpHeaders));
-    // FIXME
-    if (result.GetRawResponse().GetHeaders().find("content-range")
-        != result.GetRawResponse().GetHeaders().end())
-    {
-      ret.ContentRange = result.GetRawResponse().GetHeaders().at("content-range");
-    }
+    ret.ContentRange = result->ContentRange;
```
Comment (Jinming-Hu) [inline sdk/storage/azure-storage-files-datalake/src/datalake_file_client.cpp:329]:
```suggestion
    ret.ContentRange = std::move(result->ContentRange);
```

File: sdk/storage/azure-storage-files-datalake/inc/azure/storage/files/datalake/datalake_responses.hpp [hunk 2]
```diff
@@ -162,7 +162,8 @@ namespace Azure { namespace Storage { namespace Files { namespace DataLake { nam
   {
     std::unique_ptr<Azure::Core::Http::BodyStream> Body;
     PathHttpHeaders HttpHeaders;
-    Azure::Core::Nullable<std::string> ContentRange;
+    int64_t ContentLength = int64_t();
```
Comment (Jinming-Hu) [inline sdk/storage/azure-storage-files-datalake/inc/azure/storage/files/datalake/datalake_responses.hpp:165]: This is not mentioned in changelog.

File: sdk/storage/azure-storage-files-shares/src/share_file_client.cpp [hunk 3]
```diff
@@ -641,8 +641,7 @@ namespace Azure { namespace Storage { namespace Files { namespace Shares {
     int64_t fileRangeSize;
     if (firstChunkOptions.Range.HasValue())
     {
-      fileSize = std::stoll(firstChunk->ContentRange.GetValue().substr(
-          firstChunk->ContentRange.GetValue().find('/') + 1));
+      fileSize = firstChunk->ContentLength;
```
Comment (Jinming-Hu) [inline sdk/storage/azure-storage-files-shares/src/share_file_client.cpp:644]: What is `ContentLength`? Is it the whole file size or just the chunk we downloaded in this request? Would it be clearer if we name it `FileSize`?

File: sdk/storage/azure-storage-files-shares/inc/azure/storage/files/shares/protocol/share_rest_client.hpp [hunk 4]
```diff
@@ -772,7 +772,7 @@ namespace Azure { namespace Storage { namespace Files { namespace Shares {
       Storage::Metadata Metadata;
       int64_t ContentLength = int64_t();
```
Comment (Jinming-Hu) [inline sdk/storage/azure-storage-files-shares/inc/azure/storage/files/shares/protocol/share_rest_client.hpp:773]: Is this struct exposed to end user?

What was in `ContentLength` before this PR? I mean, when downloading partial file. Is this a breaking change?

File: sdk/storage/azure-storage-files-datalake/CHANGELOG.md [hunk 5]
```diff
@@ -10,6 +10,7 @@
 
 - Removed `GetDfsUri` in all clients since they are currently implementation details.
 - `DataLakePathClient` can no longer set permissions with `SetAccessControl`, instead, a new API `SetPermissions` is created for such functionality. Renamed the original API to `SetAccessControlList` to be more precise.
+- `ContentRange` in `PathGetPropertiesResult` and `ReadDataLakeFileResult` are now `Azure::Core::Http::Range`.
```
Comment (Jinming-Hu) [inline sdk/storage/azure-storage-files-datalake/CHANGELOG.md:13]: Why do we need `ContentRange` in `PathGetPropertiesResult`?

Comment (katmsft) [reply sdk/storage/azure-storage-files-datalake/CHANGELOG.md:13]: It is returned via REST. However, this result is not used anywhere in the convenience layer.

Comment (Jinming-Hu) [reply sdk/storage/azure-storage-files-datalake/CHANGELOG.md:13]: Is it exposed to end user? If so, I have concern about this. We should discuss how to deal with it

Comment (katmsft) [reply sdk/storage/azure-storage-files-datalake/CHANGELOG.md:13]: It is not going to be exposed to the user shortly, after I have time to make every returned result in the protocol layer in `details` namespace.

File: sdk/storage/azure-storage-files-shares/src/share_file_client.cpp (hunk 3 above)
Comment (katmsft) [reply sdk/storage/azure-storage-files-shares/src/share_file_client.cpp]: Renamed.

File: sdk/storage/azure-storage-files-shares/inc/azure/storage/files/shares/protocol/share_rest_client.hpp (hunk 4 above)
Comment (katmsft) [reply sdk/storage/azure-storage-files-shares/inc/azure/storage/files/shares/protocol/share_rest_client.hpp]: Whatever it is returned in the header.

File: sdk/storage/azure-storage-files-datalake/inc/azure/storage/files/datalake/datalake_responses.hpp (hunk 2 above)
Comment (katmsft) [reply sdk/storage/azure-storage-files-datalake/inc/azure/storage/files/datalake/datalake_responses.hpp]: Added.

File: sdk/storage/azure-storage-files-datalake/src/datalake_path_client.cpp [hunk 6]
```diff
@@ -274,7 +274,7 @@ namespace Azure { namespace Storage { namespace Files { namespace DataLake {
     Models::CreateDataLakePathResult ret;
     ret.ETag = std::move(result->ETag.GetValue());
     ret.LastModified = std::move(result->LastModified.GetValue());
-    ret.ContentLength = std::move(result->ContentLength);
+    ret.FileSize = std::move(result->ContentLength);
```
Comment (Jinming-Hu) [inline sdk/storage/azure-storage-files-datalake/src/datalake_path_client.cpp:277]: It's meaningless to `std::move` from an integer. Actually I remember some compiler or code static analysis tool will give a warning.

File: sdk/storage/azure-storage-files-datalake/src/datalake_path_client.cpp [hunk 7]
```diff
@@ -382,7 +382,7 @@ namespace Azure { namespace Storage { namespace Files { namespace DataLake {
     ret.CopyCompletedOn = std::move(result->CopyCompletedOn);
     ret.ExpiresOn = std::move(result->ExpiriesOn);
     ret.LastAccessedOn = std::move(result->LastAccessedOn);
-    ret.ContentLength = result->ContentLength;
+    ret.FileSize = result->ContentLength;
```
Comment (Jinming-Hu) [inline sdk/storage/azure-storage-files-datalake/src/datalake_path_client.cpp:385]: Be consistent.

**Review by Jinming-Hu — APPROVED** [review]

File: sdk/storage/azure-storage-files-datalake/src/datalake_path_client.cpp (hunk 6 above)
Comment (katmsft) [reply sdk/storage/azure-storage-files-datalake/src/datalake_path_client.cpp:277]: It is a nullable object.

**Pull request merged**